src/main/scala/org/apache/mesos/chronos/scheduler/api/DependentJobResource.scala

```diff
@@ -57,10 +57,16 @@ class DependentJobResource @Inject()(
         val parents = jobGraph.parentJobs(newJob)
         oldJob match {
           case j: DependencyBasedJob =>
-            val oldParents = jobGraph.parentJobs(j)
-            oldParents.map(x => jobGraph.removeDependency(x.name, oldJob.name))
+            val newParentNames = parents.map(_.name)
+            val oldParentNames = jobGraph.parentJobs(j).map(_.name)
+
+            if (newParentNames != oldParentNames) {
+              oldParentNames.foreach(jobGraph.removeDependency(_, oldJob.name))
+              newParentNames.foreach(jobGraph.addDependency(_, newJob.name))
+            }
             jobScheduler.removeSchedule(j)
           case j: ScheduleBasedJob =>
```

Comment:
No, that is not necessary, because a former `ScheduleBasedJob` has no parents/dependencies.
Also it is not possible as `JobGraph.parentJobs` only accepts values of type `DependencyBasedJob`.